pkg/asset/installconfig/aws/permissions.go

```diff
@@ -316,6 +319,9 @@ var permissions = map[PermissionGroup][]string{
 		"kms:CreateGrant",
 		"kms:ListGrants",
 	},
+	PermissionAMIEncryptionKeys: {
```

Comment:
What about to enhance the documentation to specify scenarios which this group is required? It would help in the OCP and managed services documentation (ROSA needs detailed needs on managed IAM roles creation/update).
Something like the following suggestion.
```suggestion
    // PermissionAMIEncryptionKeys are IAM permissions when using Encrypted AMIs, on scenarios like:
    // - BYO KMS
    // - BYO KMS and BYO AMI
    // - BYO Encrypted AMI
	PermissionAMIEncryptionKeys: {
```

Reply:
I like this idea, but I think the comments should be added where we define the PermissionGroup constants. And I think the existing comment may already cover this, happy for suggestions on the wording...

I have kept this AMI permission (only `kms:ReEncrypt*`) separate from BYO KMS, because apparently BYO KMS does not need this permission.

So the only scenario where this specific permission is used is BYO (Encrypted) AMI. Currently this PR cannot distinguish an encrypted AMI from an unencrypted AMI, so the permissions are required anytime a custom AMI is used. I will discuss more in a separate comment.

Reply:
> It would help in the OCP and managed services documentation (ROSA needs detailed needs on managed IAM roles creation/update).
Something like the following suggestion.

I think the doc should be consistent with the installer behaviors, even if unused permissions are introduced.